src/main/java/bio/terra/service/load/LoadDao.java

```diff
@@ -49,72 +44,101 @@ public LoadDao(NamedParameterJdbcTemplate jdbcTemplate, ConfigurationService con
+                    .lockingFlightId(rs.getString("locking_flight_id")));
+
+            if (load == null) {
+                throw new CorruptMetadataException("Load row should exist! Load tag: " + loadTag);
             }
-        }
 
-        try {
-            // Lock the load for our use and validate
-            updateLoad(load.getId(), true, flightId);
-        } catch (CannotSerializeTransactionException ex) {
-            conflictThrow(load);
+            // It is locked by someone else
+            if (!StringUtils.equals(load.getLockingFlightId(), flightId)) {
+                throw new LoadLockedException("Load '" + loadTag +
+                    "' is locked by flight '" + load.getLockingFlightId() + "'");
+            }
+        } else {
+            load = new Load()
+                .id(keyHolder.getId())
+                .loadTag(keyHolder.getString("load_tag"))
+                .locked(keyHolder.getField("locked", Boolean.class))
+                .lockingFlightId(keyHolder.getString("locking_flight_id"));
         }
 
-        load = lookupLoadByTag(loadTag);
-        if (load != null && load.isLocked() && StringUtils.equals(load.getLockingFlightId(), flightId)) {
-            return load;
-        }
-        throw new LoadLockFailureException("Internal error: failed to lock a load!");
+        return load;
     }
 
-    // To support idempotent steps, this method will not complain if the load no longer exists or if
-    // it is already unlocked. It throws if the lock is held by a different flight or if the unlock operation
-    // fails for some reason.
+    /**
+     * Unlocking means deleting the specific row for our load tag and flight id.
+     * If no row is deleted, we assume this is a redo of the unlock and is OK.
+     * If a row is deleted, then we have performed the unlock. So we don't
+     * check the affected row count.
+     *
+     * @param loadTag tag identifying this load
+     * @param flightId flight id releasing the lock
+     */
     @Transactional(propagation = Propagation.REQUIRED, isolation = Isolation.SERIALIZABLE)
     public void unlockLoad(String loadTag, String flightId) {
-        Load load = lookupLoadByTag(loadTag);
-        if (load == null || !load.isLocked()) {
-            return; // nothing to unlock
-        }
+        jdbcTemplate.getJdbcTemplate().execute("LOCK TABLE load IN EXCLUSIVE MODE");
 
-        if (!StringUtils.equals(load.getLockingFlightId(), flightId)) {
-            conflictThrow(load);
-        }
-        updateLoad(load.getId(), false, null);
-        load = lookupLoadByTag(loadTag);
-        if (load == null || load.isLocked() || load.getLockingFlightId() != null) {
-            throw new LoadLockFailureException("Internal error: failed to unlock a load!");
+        String delete = "DELETE FROM load WHERE load_tag = :load_tag AND locking_flight_id = :flight_id";
+        MapSqlParameterSource params = new MapSqlParameterSource()
```

Comment:
it seems like we are often looking up values based on load_tag over any other col, is that true? Could we optimize around that?